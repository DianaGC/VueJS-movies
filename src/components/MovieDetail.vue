<template>
 <v-container>
   <v-layout wrap>
     <v-flex xs12 mr-1 ml-1>
       <v-card>
         <v-img contain
                :src="movieDetail.Poster"
                aspect-ratio="2"></v-img>
         <v-card-text primary-title>
           <div>
             <h1 class="headline mb-0">{{movieDetail.Title}}-{{movieDetail.Year}}</h1>
             <p>{{ movieDetail.Plot}} </p>
             <h3>Actors: </h3>{{movieDetail.Actors}}
             <h3>Awards: </h3> {{movieDetail.Awards}}
             <h3>Genre: </h3> {{movieDetail.Genre}}
           </div>
         </v-card-text>
         <v-card-actions>
           <v-btn flat color="green" v-on:click="getBack">back</v-btn>
         </v-card-actions>
       </v-card>
     </v-flex>
   </v-layout>

   <v-layout  wrap>
     <v-flex xs12>
       <div class="justify-center" >
         <v-dialog  v-model="dialog" width="500">
           <v-btn  slot="activator"
                   color="black"
                   @click.stop="dialog = true"
                   dark>
             View Ratings
           </v-btn>
           <v-card>
             <v-card-title class="headline grey lighten-2" primary-title>
               Ratings
             </v-card-title>
             <v-card-text>
               <table style="width:100%" >
                 <tr>
                   <th>Source</th>
                   <th>Ratings</th>
                 </tr>
                 <tr v-for="(rating, index) in ratings" :key="index">
                   <td align="center">{{rating.Source}}</td>
                   <td align="center">
                     <v-rating :half-increments="true" :value="rating.Value"></v-rating>
                   </td>
                 </tr>
               </table>
             </v-card-text>
             <v-divider></v-divider>
             <v-card-actions>
               <v-spacer></v-spacer>
               <v-btn
                       color="primary"
                       flat
                       v-on:click="dialog = false"
               >

                 OK
               </v-btn>
             </v-card-actions>
           </v-card>
         </v-dialog>
       </div>
     </v-flex>
   </v-layout>
   <Comments></Comments>
 </v-container>

</template>

<script>
import {getMovieDetail} from '@/service/movies'
import Comments from "./Comments";

export default {
  components: {Comments},
  props: ['id'],
  data () {
    return {
      movieDetail: '',
      ratings: '',
      dialog: false
    }
  },
  mounted () {
    getMovieDetail(this.id)
      .then(response => {
        this.movieDetail = response;
        this.ratings = response.Ratings;
        this.ratings.forEach(function (element) {
          element.Value = parseFloat(element.Value)
          element.Value = element.Value <= 10 ? element.Value / 2 : element.Value / 20
        })
      })
      .catch(e => console.log(e))
  },
  methods: {
    getBack () {
      this.$router.push('/')
    }
  }
}
</script>
