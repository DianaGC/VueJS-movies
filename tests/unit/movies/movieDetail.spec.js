import Vue from 'vue'
import Vuetify from 'vuetify'
import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import MovieDetail from "../../../src/components/MovieDetail";

const localVue = createLocalVue();
localVue.use(Vuetify);

describe("Component MovieDetail.vue", () => {
  let vuetify = new Vuetify()
  beforeEach(() =>{
    vuetify = new Vuetify()
  });
  it('should render ClientModal page', () => {
    const wrapper = mount(MovieDetail);
    expect(wrapper).toBeDefined();
  });

  it('Matches the snapshot', () => {
    const wrapper = mount(MovieDetail, {
      localVue,
      vuetify,
      propsData: {
        movieDetail: '',
        ratings: '',
        dialog: false
      },
    });
    expect(wrapper.html()).toMatchSnapshot()
  });

  it("should emit an event when the action v-btn is clicked", () => {
    const id = "tt0087814";
    const wrapper = mount(MovieDetail, {
      localVue,
      vuetify,
      propsData: {
        movieDetail: '',
        ratings: '',
        dialog: false
      },
    });
    const event = jest.fn();
    const button = wrapper.find('.v-btn');
    wrapper.vm.$on('getBack', event);
    button.trigger('click');
    expect(event).toHaveBeenCalled()
  });

});


