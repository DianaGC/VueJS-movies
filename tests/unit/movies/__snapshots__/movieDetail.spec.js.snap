// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MovieDetail.vue Matches the snapshot 1`] = `
<v-container moviedetail="" ratings="">
  <v-layout wrap="">
    <v-flex xs12="" mr-1="" ml-1="">
      <v-card>
        <v-img contain="" aspect-ratio="2"></v-img>
        <v-card-text primary-title="">
          <div>
            <h1 class="headline mb-0">-</h1>
            <p> </p>
            <h3>Actors: </h3>
            <h3>Awards: </h3>
            <h3>Genre: </h3>
          </div>
        </v-card-text>
        <v-card-actions>
          <v-btn flat="" color="green">back</v-btn>
        </v-card-actions>
      </v-card>
    </v-flex>
  </v-layout>
  <v-layout wrap="">
    <v-flex xs12="">
      <div class="justify-center">
        <v-dialog width="500">
          <v-btn slot="activator" color="black" dark="">
            View Ratings
          </v-btn>
          <v-card>
            <v-card-title primary-title="" class="headline grey lighten-2">
              Ratings
            </v-card-title>
            <v-card-text>
              <table style="width: 100%;">
                <tr>
                  <th>Source</th>
                  <th>Ratings</th>
                </tr>
              </table>
            </v-card-text>
            <v-divider></v-divider>
            <v-card-actions>
              <v-spacer></v-spacer>
              <v-btn color="primary" flat="">

                OK
              </v-btn>
            </v-card-actions>
          </v-card>
        </v-dialog>
      </div>
    </v-flex>
  </v-layout>
</v-container>
`;
